tests/wallet/cat_wallet/test_offer_lifecycle.py

```diff
@@ -47,24 +48,22 @@ def str_to_cat_hash(tail_str: str) -> bytes32:
 
 # This method takes a dictionary of strings mapping to amounts and generates the appropriate CAT/XCH coins
 async def generate_coins(
-    sim,
-    sim_client,
-    requested_coins: Dict[Optional[str], List[uint64]],
+    sim: SpendSim, sim_client: SimClient, requested_coins: Dict[Optional[str], List[int]]
 ) -> Dict[Optional[str], List[Coin]]:
     await sim.farm_block(acs_ph)
-    parent_coin: Coin = [cr.coin for cr in await sim_client.get_coin_records_by_puzzle_hash(acs_ph)][0]
+    parent_coin = [cr.coin for cr in await sim_client.get_coin_records_by_puzzle_hash(acs_ph)][0]
 
     # We need to gather a list of initial coins to create as well as spends that do the eve spend for every CAT
-    payments: List[Payment] = []
-    cat_bundles: List[SpendBundle] = []
+    payments = []
+    cat_bundles = []
```

Comment:
I think that being explicit in this context is often good and makes it easier to read since the append etc require scanning further down to figure out what this is going to become.

Reply:
Yep, I tend to opt for this only when the current block itself makes it clear what we're appending (in this example `payments.append(Payment(...))`).

In general, I found so many instances where we declare the wrong types, or shadow values later on but with different types, so whenever the type is clear from its immediate use I favor that.